applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_central_differences_scheme.hpp

```diff
@@ -0,0 +1,727 @@
+
+
+        for (SizeType j = 0; j < DoF; j++)
+        {
+            if (fix_displacements[j])
+            {
+              r_current_acceleration[j]  = 0.0;
+              r_middle_velocity[j]       = 0.0;
+            }
+
+          r_middle_velocity[j]       = 0.0 + (mTime.Middle - mTime.Previous) * r_current_acceleration[j] ;
+          r_current_velocity[j]      = r_middle_velocity[j] + (mTime.Previous - mTime.PreviousMiddle) * r_current_acceleration[j]; //+ actual_velocity;
+          //r_current_displacement[j]  = 0.0;
+
+
+
+        }//for DoF
+        ////// ROTATION DEGRESS OF FREEDOM
+        if (has_dof_for_rot_z)
+        {
+
+          const array_1d<double,3>& nodal_inertia                       = (i_begin+i)->GetValue(NODAL_INERTIA);
+          array_1d<double,3>& r_current_residual_moment          = (i_begin+i)->FastGetSolutionStepValue(MOMENT_RESIDUAL);
+          array_1d<double,3>& r_current_angular_velocity         = (i_begin+i)->FastGetSolutionStepValue(ANGULAR_VELOCITY);
+          //array_1d<double,3>& current_rotation                 = i_begin->FastGetSolutionStepValue(ROTATION);
+          array_1d<double,3>& r_middle_angular_velocity          = (i_begin+i)->GetValue(MIDDLE_ANGULAR_VELOCITY);
+          array_1d<double,3>& r_current_angular_acceleration     = (i_begin+i)->FastGetSolutionStepValue(ANGULAR_ACCELERATION);
+
+
+          for (SizeType kk = 0; kk<DoF; ++kk)
+          {
+            if (nodal_inertia[kk] > numerical_limit)  r_current_angular_acceleration[kk] = r_current_residual_moment[kk] / nodal_inertia[kk];
+            else r_current_angular_acceleration[kk] = 0.00;
+          }
+
+          DoF = 3;
+          bool fix_rotation[3] = {false, false, false};
+          fix_rotation[0] = ((i_begin+i)->pGetDof(ROTATION_X))->IsFixed();
+          fix_rotation[1] = ((i_begin+i)->pGetDof(ROTATION_Y))->IsFixed();
+          fix_rotation[2] = ((i_begin+i)->pGetDof(ROTATION_Z))->IsFixed();
+
+
+
+          for (SizeType j = 0; j < DoF; j++)
+          {
+            if (fix_rotation[j])
+            {
+              r_current_angular_acceleration[j] = 0.00;
+              r_middle_angular_velocity[j] = 0.00;
+            }
+
+            r_middle_angular_velocity[j]   = 0.00 + (mTime.Middle - mTime.Previous) * r_current_angular_acceleration[j] ;
+            r_current_angular_velocity[j]  = r_middle_angular_velocity[j] + (mTime.Previous - mTime.PreviousMiddle) * r_current_angular_acceleration[j];
+            //current_rotation[j]          = 0.00;
+          }//trans DoF
+        }// Rot DoF
+      }//for node parallel
+
+
+      mTime.Previous = mTime.Current;
+      mTime.PreviousMiddle = mTime.Middle;
+      KRATOS_CATCH("")
+    }
+
+
+  //***************************************************************************
+  //***************************************************************************
+  void Calculate_RHS_Contribution(Element::Pointer pCurrentElement,
```

Comment:
the content of this and the next function should be templated, i.e. both functions call the same (templated for elems/conds) function

You do different things inside these functins even though I guess you should be doing the same
(for elements you call `CalculateLocalSystem` and for conditions you call `CalculateRightHandSide`)!